tests/acceptance/features/bootstrap/SpacesContext.php

```diff
@@ -463,6 +457,38 @@ public function deleteAllSpacesOfTheType(string $driveType): void {
 		}
 	}
 
+	/**
+	 * users delete their personal space at the end of the test
+	 *
+	 * @return void
+	 *
+	 * @throws Exception|GuzzleException
+	 */
+	public function deleteAllPersonalSpaces(): void {
+		$query = "\$filter=driveType eq personal";
+		$createdUsers= $this->featureContext->getCreatedUsers();
+
+		foreach($createdUsers as $user)
+		{
+			for ($i = 0; $i < 2; ++$i) {
+
+				$this->theUserListsAllHisAvailableSpacesUsingTheGraphApi(
+					$user["actualUsername"],
+					$query
+				);
+				$drives = $this->getAvailableSpaces();
+
+				foreach ($drives as $value) {
+					if (!\array_key_exists("deleted", $value["root"])) {
+						$this->sendDisableSpaceRequest($user["actualUsername"], $value["name"]);
```

Comment:
```suggestion
						$this->sendDisableSpaceRequest($user["actualUsername"], $value["name"]);
						$this->sendDeleteSpaceRequest($user["actualUsername"], $value["name"]);
```

then we can save the `for ($i = 0; $i < 2; ++$i` loop

Reply:
I refactored.